src/server/service/kv.rs

```diff
@@ -112,11 +117,31 @@ impl<T: RaftStoreRouter + 'static, E: Engine, L: LockManager> Service<T, E, L> {
         let status = RpcStatus::new(code, Some(format!("{}", err)));
         ctx.spawn(sink.fail(status).map_err(|_| ()));
     }
+
+    fn check_common_names(&self, ctx: &RpcContext) -> bool {
+        if self.x509_common_names.is_empty() {
+            return true;
+        } else if let Some(auth_context) = ctx.auth_context() {
```

Comment:
If secure channel is not used, the check should be skipped.

Reply:
Fix